Kitty_Paw_Keypad/code.py

```diff
@@ -52,89 +57,62 @@
+parrot0_grid = displayio.TileGrid(bitmap, pixel_shader=palette,
+                                 tile_height=32, tile_width=32)
 
 # Create a Group to hold the TileGrid
-group = displayio.Group()
+group = displayio.Group(scale=4, x = 64, y = 32)
 
 # Add the TileGrid to the Group
 group.append(parrot0_grid)
 
 # Add the Group to the Display
 display.show(group)
 
-#  digital pins for the buttons
-key_pins = [board.A0, board.A1, board.A2, board.A3]
-
-#  array for buttons
-keys = []
+#  setup button pins
+key_pins = (
+    board.A0,
+    board.A1,
+    board.A2,
+    board.A3,
+)
 
-#  setup buttons as inputs
-for key in key_pins:
-    key_pin = digitalio.DigitalInOut(key)
-    key_pin.direction = digitalio.Direction.INPUT
-    key_pin.pull = digitalio.Pull.UP
-    keys.append(key_pin)
+#  create keypad
+keys = keypad.Keys(key_pins, value_when_pressed=False, pull=True)
 
 p = 0 #  variable for tilegrid index
-a = 0 #  variable for tile position
-
-#  states for buttons
-key0_pressed = False
-key1_pressed = False
-key2_pressed = False
-key3_pressed = False
-
-#  array for button states
-key_states = [key0_pressed, key1_pressed, key2_pressed, key3_pressed]
 
-last_p = p
 while True:
-    #  default tile grid position
-    if last_p != p:
-        parrot0_grid[a] = p
-        last_p = p
-
-    #  iterate through 4 buttons
-    for i in range(4):
-        inputs = keys[i]
-        #  if button is pressed...
-        if not inputs.value and key_states[i] is False:
-            #  tile grid advances by 1 frame
+    #  update parrot sprite
+    parrot0_grid[0] = p
+
+    #  get keypad inputs
+    event = keys.events.get()
+    if event:
+        #  if a key is pressed..
+        if event.pressed:
+            #  increase sprite index by 1
             p += 1
-            #  update button state
-            key_states[i] = True
-            #  if a midi keyboard...
+            #  if a midi keyboard
             if midi_mode:
-                #  send NoteOn for corresponding MIDI note
-                midi.send(NoteOn(midi_notes[i], 120))
-            #  if an HID keyboard...
+                #  send note number
+                midi.send(NoteOn(midi_notes[event.key_number], 120))
+            #  if hid keyboard
             if keyboard_mode:
-                #  send keyboard output for corresponding keycode
-                #  the default includes a modifier along with the keycode
-                keyboard.send(ctrl, shortcuts[i])
-            #  if the tile grid's index is at 9...
-            if p > 9:
-                #  reset the index to 0
-                p = 0
-        #  if the button is released...
-        if inputs.value and key_states[i] is True:
-            #  update button state
-            key_states[i] = False
-            #  if a midi keyboard...
+                #  send hid keyboard shortcut
+                keyboard.send(ctrl, shortcuts[event.key_number])
+        #  if a key is released
+        if event.released:
```

Comment:
Move increment and picture advance to here.
```suggestion
                #  send hid keyboard shortcut
                keyboard.send(ctrl, shortcuts[event.key_number])
            # advance to next parrot pic; wrap around at end.
            p = (p + 1) % 10
            #  update parrot sprite
            parrot0_grid[0] = p        #  if a key is released
        if event.released:
```

Reply:
yesss excellent, will do!